src/common/overlay/overlay.ts

```diff
@@ -6,31 +8,52 @@ export interface OverlayOptions {
 
 export interface OverlayShowOptions {
     onClick?(event: MouseEvent): void;
+    onClickClose?(event: MouseEvent): void;
+}
+
+export interface Elements {
+    element: HTMLElement;
+    styles: HTMLElement;
+    closeElement?: HTMLElement;
+    layoutElement?: HTMLElement;
 }
 
 export default class Overlay {
     private _element: HTMLElement;
+    private _styles: HTMLElement;
+    private _closeElement?: HTMLElement;
+    private _layoutElement?: HTMLElement;
     private _unregisterClick?: () => void;
 
-    constructor(options?: OverlayOptions) {
-        this._element = this._createElement(options);
+    constructor(options?: OverlayOptions, isCloseElement?: boolean, innerHtml?: HTMLElement | HTMLElement[]) {
+        const { styles, element, closeElement, layoutElement } = this._createElements(options, isCloseElement, innerHtml);
+
+        this._element = element;
+        this._styles = styles;
+        this._closeElement = closeElement;
+        this._layoutElement = layoutElement;
     }
 
     show(options?: OverlayShowOptions): void {
-        if (this._element.parentElement) {
+        const element = this._layoutElement || this._element;
+
+        if (element.parentElement) {
             return;
         }
 
         this._registerClick(options);
 
-        document.body.appendChild(this._element);
+        document.body.appendChild(element);
+        document.body.appendChild(this._styles);
 
         // Fade In
         setTimeout(() => this._element.style.opacity = '1');
     }
 
     remove(): void {
-        if (!this._element.parentElement) {
+        const element = this._layoutElement || this._element;
```

Comment:
I understand your doubt. But we need both components. In this case we have `element` (it's darkened div with click event) and `layoutElement` (it needs only for wrapper `element` and `closeElement`), that  are needed for different purpose. If it would be one element we add event listener to `layoutElement ` and click on the `closeElement` won't work

In `remove()` we should remove `element` (if overlay without closeElement) or `layoutElement` (if overlay with closeElement), so I decided to create `mainElement`, that will be `element` or `layout` and remove `styleElement` (I added it to `mainElement`)